### ISSUE TITLE
Callback for succesful logging

### ISSUE THREAD

**Issue by dotmilk** [rapid7/le_node#77]
For instance i have a worker, who just lost connection to redis. I want the process to exit after logging. But if I
```
log.crit('worker x died:' + err)
process.exit(1)
```
I never get the log entry, and if i wrap it in a setTimeout or some similar construct, there is the chance that it gets called multiple times as many requests to redis will be failing sequentially now, but i don't care about them as they are noise. So a callback would eliminate this problem and i could exit immediately after i know my entry is safe. 

Perhaps this feature is already here somehow and i am over looking it.

**Comment by bathos** [comment]
By the nature of how logs are sent over tcp, it's unfortunately not possible to bind a callback to a *specific* log entry having been committed. The TCP connection is just a one-way continuous stream of text -- entries are delimited by newlines, and any given chunk being sent may include portions of one or more entries -- and we never hear anything back.

However I believe there's a way to handle the case you've described, something like this:

    let somethingWentWrong = false;

    // later, in the logic for exiting on error...
    if (somethingWentWrong) return;
    somethingWentWrong = true;
    log.crit('worker x died:' + err);
    log.on('finish', () => process.exit(1));
    log.end();

Since the logger is in the worker, calling `end()` on the stream should be okay -- and then you're just listening for the stream to say it's done sending everything. However I'm not 100% sure this will work because there are really two streams involved. If this solution doesn't work, I'll have to dig in a bit -- the logic for closing the stream was never really as fleshed out as it ought to be.

**Comment by cheshirecatalyst** [comment]
Hmm that seems to be an awesome take, however :( `Error: write after end` is the outcome, it basically loops through there never actually sending to le and ending the process.

**Comment by bathos** [comment]
If you got a Write After End error, you still have handlers in there not obeying 'somethingWentWrong' after it's been set to true (you know, on reflection, I think writable streams have an `ended` property you can check instead of having a flag like this -- definitely a nicer approach). It may be that the finish event is firing before the network stream is actually terminated, but it also could be that the error caused by writing after end() is responsible. I'll try some tests over here soon though since what you've described should be easy to replicate.

**Comment by cheshirecatalyst** [comment]
```
    let errorThrown

    process.on('uncaughtException',finalErr)

    function finalErr(err){

        if (!errorThrown) {
            errorThrown = true
            console.log('final',err)
            log.crit("Worker dying: " + err)
            log.on('finish', function(){process.exit(1)})
            log.end()
        }

    }

    require('./worker')
```

```
final [Error: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED 127.0.0.1:6379]
[Error: write after end]
```

**Comment by gearoidof-r7** [comment]
This has been recently fixed in version 1.6.3
